pkg/flink/internal/controller/input_controller.go

```diff
@@ -31,7 +34,28 @@ type InputController struct {
 
 const defaultWindowSize = 100
 
-func NewInputController(history *history.History, lspCompleter prompt.Completer) types.InputControllerInterface {
+func NewLSPHandler(c types.InputControllerInterface, handlerCh chan *jsonrpc2.Request) {
```

Comment:
I think I'd rename this to something like `startLspRequestHandler` or something like that

Reply:
done